### DISCUSSION TITLE
Documentation pour la modification de la documentation

### DISCUSSION THREAD

**Discussion by xylle** [sqlpage/SQLPage#697] [General]
Bonjour,
aurais-tu une documentation pour modifier la documentation ?

Je me suis aperçue qu'il manque de la documentation pour le bouton réinitialiser du formulaire.
Mais comme j'ai trouvé vraiment pénible d'ajouter la documentation pour l'option vide du select, j'hésite à le faire.

J'ai modifié le formulaire pour ajouter un bouton `revenir en arrière` et j'ai groupé les trois boutons avec btn-list. J'ai aussi ajouté une option pour aligner le groupe de bouton. Est-ce quelque chose qui t'intéresse. Si oui, je fais une demande d'extraction (pull request)
```
        <div class="btn-list justify-content-{{#if btn_list_justify}}{{btn_list_justify}}{{else}}start{{/if}}">
        {{#if (ne validate '')}} 
            <input class="btn
            btn-{{default validate_color "primary"}}
            {{#if validate_shape}} btn-{{validate_shape}}{{/if}}
            {{#if validate_outline}} btn-outline-{{validate_outline}}{{/if}}
            {{#if validate_size}} btn-{{validate_size}}{{/if}}"
            {{flush_delayed}}
            type="submit"
            {{#if validate}}value="{{validate}}"{{/if}}>
        {{/if}}
        {{#if reset}}
            <input class="btn
            {{#if reset_color}} btn-{{reset_color}}{{else}} btn-white{{/if}}
            {{#if reset_shape}} btn-{{reset_shape}}{{/if}}
            {{#if reset_outline}} btn-outline-{{reset_outline}}{{/if}}
            {{#if reset_size}} btn-{{reset_size}}{{/if}}"
            type="reset" 
            value="{{reset}}">
        {{/if}}
        {{#if go_back}}
            <input class="btn
            {{#if go_back_color}} btn-{{go_back_color}}{{else}} btn-red{{/if}}
            {{#if go_back_shape}} btn-{{go_back_shape}}{{/if}}
            {{#if go_back_outline}} btn-outline-{{go_back_outline}}{{/if}}
            {{#if go_back_size}} btn-{{go_back_size}}{{/if}}"
            id="go_back"
            value="{{go_back}}">
        {{/if}}
        </div>

```

**Comment by xylle** [comment]
Pour que ça fonctionne, il faut ajouter du JS : 
```
document.addEventListener('DOMContentLoaded', function() {
    var button = document.getElementById('go_back');
    if (button) {
        button.addEventListener('click', function() {
            window.history.back();
        });
    }
});
```
On doit sans doute pouvoir faire mieux.

**Reply by lovasoa** [reply]
Oui, cette approche ne fonctionnera pas si il y a plusieurs boutons go_back sur une page.

Mais de manière plus générale, quel est le cas d'utilisation ? Dans quel cas est-ce que tu ne peux pas connaître la page dont vient l'utilisateur, pour faire un bouton normal ? Et au pire, on a toujours `sqlpage.header('referer')`...

Est-ce que tu peux expliquer ici un peu l'application que tu es en train de construire ? Je suis toujours curieux ! :slightly_smiling_face:

**Reply by xylle** [reply]
Même s'il y a plusieurs boutons go_back sur la page, ça marchera. Il va ajouter l'événement sur tous les boutons et comme il n'y a qu'une seule page précédente dans l'historique, il retournera sur la bonne. Normalement ça devrait faire exactement la même chose avec le `document.referrer`, il n'y en a qu'un. Il faudrait peut-être éviter d'utiliser l'id afin de respecter l'unicité demander par la norme.

Quand je fais un formulaire html, j'aime bien avoir le bouton annuler, s'il n'y est pas, ça manque.
En gros, je ne sais pas ce que j'ai modifié, cliquer sur annuler me rassure. Réinitialiser ne fait pas la même chose, il remet le formulaire à l'état initial, mais on peut encore faire n'importe quoi. Qui n'a pas fait rouler la mollette de souris sur un select et a enregistré le changement, sans faire attention ?
De plus, je répète à mes apprentis que si tu n'as pas voulu modifier une page, il ne faut pas cliquer sur enregistrer/modifier, mais sur annuler. Ça évite les erreurs d'inattention.

On pourrait mettre un bouton avec un lien définit par le développeur, mais ça complexifie, je trouve.

Pareil pour les boutons à gauche, ce n'est pas standard. Je comprends que ça plait, mais mettre une option qui permet de les positionner comme le développeur le veut, ça ne coute rien.

Je fais une application pour la gestion de machines virtuelles dans vcenter avec ansible.
- Création de vm à partir d'un modèle. Le modèle doit respecter des prérequis.
- Suppression/Décommissionnement de VM
- Configuration du réseau
- Application de rôle ansible
- Agrandissement des partitions

**Comment by lovasoa** [comment]
Pour la documentation: toute la documentation pour les développeurs est ici: https://github.com/sqlpage/SQLPage/blob/main/CONTRIBUTING.md

Elle n'est pas parfaite, si tu veux la relire et proposer des modifications dessus, c'est bienvenue !
